git-svn revision support

kfb opened the pull request:
I do a lot of reviewing of code hosted in an SVN repo; developers will often reference the SVN revision number of changes. This patch adds another column to the history view that shows the SVN revision number, extracted from the `git-svn-id` line. This allows for quickly mapping between git and SVN commits.

In quick testing, performance doesn't seem to be heavily impacted, but if others find differently it could be an off-by-default option.

(Apologies also for the XIB diff spam, I'm never sure which of the bits IB changes are actually required, so I played it safe and committed the whole thing)

Commit:
Extract SVN revision number from git log

This patch adds parsing of the output of git log to extract the SVN revision
number tied to each commit. The revision number is part of the git-svn-id
string.

Commit:
Update GUI to show SVN revision

rowanj commented:
I haven't reviewed this yet; in the way of that is that I don't have a good git-svn repo to test with (at hand, I do have archives laying about).

Don't worry about the XIBs.  They don't scare me.

Ideally, I think the column should be hidden (unavailable) unless the data is actually present; and probably not shown by default unless there is actually a git-svn remote configured; as in my experience, the commits survive indefinitely after a project has migrated entirely to git.

Commit:
Add hasSVNRemote to PBGitRepository

This patch adds the ability to query the repo for the presence of any
SVN remotes. This is determined by whether the string "svn-remote" is
present in the repository's .git/config file.

Commit:
Only load SVN revisions if there's an SVN remote

Commit:
Hide the SVN rev column if no SVN remote

This patch causes the "Git SVN Revision" table column to be removed
when a repository has no SVN remote.

Commit:
XIB updates

kfb commented:
Okay, so this should bring the pull request up to date with your comments. I've rebased against the current master and added the auto-hiding of the column for repositories with no SVN remote configured. I think there's probably a better way to go about this, but I'm not familiar enough with GitX's source to know if that's true or not.

rowanj commented:
Commits look good; I'll just run it over my test repos and this should be good to make it into the next build.

rowanj commented:
- Doesn't seem to find the svn revision in commits that were made without a message

rowanj commented:
@kfb FYI, Objective-Git does provide a facility for reading all remote properties; it's just through the GTConfiguration, not GTRemote; see my changes in PBGitRepository.m if you're curious :-)
